pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -291,6 +367,56 @@ func (c *Cluster) runJob(job *batch.Job, nodeName string, config *provisionConfi
 	return true
 }
 
+func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig, configMap *v1.ConfigMap, status *OrchestrationStatus) {
+	osds := status.OSDs
+	logger.Infof("starting %d osd daemons on node %s", len(osds), pvcName)
+	conf := make(map[string]string)
+	storeConfig := osdconfig.ToStoreConfig(conf)
+	osdProps := osdProperties{
+		crushHostname: pvcName,
+		pvc: v1.PersistentVolumeClaimVolumeSource{
+			ClaimName: pvcName,
+		},
+	}
+
+	// start osds
+	for _, osd := range osds {
+		logger.Debugf("start osd %v", osd)
+		dp, err := c.makeDeployment(osdProps, osd)
+		if err != nil {
+			errMsg := fmt.Sprintf("failed to create deployment for node %s: %v", osdProps.crushHostname, err)
+			config.addError(errMsg)
+			continue
+		}
+
+		_, err = c.context.Clientset.AppsV1().Deployments(c.Namespace).Create(dp)
```

Comment:
From here to the end of the method we should be able to factor this out and call a common method for creating an osd deployment. No need to duplicate this here, right? The code for the other types of osds looks the same for creating the deployment.

Reply:
cc @ashishranjan738

Reply:
As per @ashishranjan738 we will take it in the next PR as it only involves refactoring